src/EditorFeatures/Core/Wrapping/SeparatedSyntaxList/SeparatedSyntaxListCodeActionComputer.cs

```diff
@@ -0,0 +1,376 @@
+
+                var indentString = afterOpenTokenOffset.CreateIndentationString(UseTabs, TabSize);
+                return indentString;
+            }
+
+            private string GetSingleIdentation()
+            {
+                // Insert a newline after the open token of the list.  Then ask the
+                // ISynchronousIndentationService where it thinks that the next line should be
+                // indented.
+                var openToken = _listSyntax.GetFirstToken();
+
+                var newSourceText = OriginalSourceText.WithChanges(new TextChange(new TextSpan(openToken.Span.End, 0), NewLine));
+                newSourceText = newSourceText.WithChanges(
+                    new TextChange(TextSpan.FromBounds(openToken.Span.End + NewLine.Length, newSourceText.Length), ""));
+                var newDocument = OriginalDocument.WithText(newSourceText);
+
+                var originalLineNumber = newSourceText.Lines.GetLineFromPosition(openToken.Span.Start).LineNumber;
+                var desiredIndentation = _indentationService.GetBlankLineIndentation(
+                    newDocument, originalLineNumber + 1,
+                    FormattingOptions.IndentStyle.Smart,
+                    CancellationToken);
+
+                var baseLine = newSourceText.Lines.GetLineFromPosition(desiredIndentation.BasePosition);
+                var baseOffsetInLine = desiredIndentation.BasePosition - baseLine.Start;
+
+                var indent = baseOffsetInLine + desiredIndentation.Offset;
+
+                var indentString = indent.CreateIndentationString(UseTabs, TabSize);
+                return indentString;
+            }
+
+            private SyntaxTrivia GetIndentationTrivia(WrappingStyle wrappingStyle)
+            {
+                return wrappingStyle == WrappingStyle.UnwrapFirst_AlignRest
+                    ? _afterOpenTokenIndentationTrivia
+                    : _singleIndentationTrivia;
+            }
+
+            protected override async Task<ImmutableArray<WrappingGroup>> ComputeWrappingGroupsAsync()
+            {
+                var result = ArrayBuilder<WrappingGroup>.GetInstance();
+                await AddWrappingGroups(result).ConfigureAwait(false);
+                return result.ToImmutableAndFree();
+            }
+
+            private async Task AddWrappingGroups(ArrayBuilder<WrappingGroup> result)
+            {
+                result.Add(await GetWrapEveryGroupAsync().ConfigureAwait(false));
+                result.Add(await GetUnwrapGroupAsync().ConfigureAwait(false));
+                result.Add(await GetWrapLongGroupAsync().ConfigureAwait(false));
+            }
+
+            #region unwrap all
+
+            private async Task<WrappingGroup> GetUnwrapGroupAsync()
+            {
+                var unwrapActions = ArrayBuilder<WrapItemsAction>.GetInstance();
+
+                var parentTitle = string.Format(FeaturesResources.Unwrap_0, Wrapper.ListName);
```

Comment:
❕ We can keep this for now, but we should be prepared to create a pull request that expands these as single resources if one or more languages cannot represent it with correct grammar.

/cc @tmeschter @nguerrera in case they have additional information to say this either definitely is or definitely is not a problem.

Reply:
It probably will be an issue. I'll proactively make the change.

Reply:
It's not easy to see what the potential issue is. Is this composed with other resources somewhere else?

Reply:
Hey Nick.  The primary issue that I'm concerned with is that we are offering a single feature that helps to wrap multiple types of constructs.  i.e. it can wrap parameters.  it can wrap arguments.  it can wrap expressions.  There are several strings we show the user here, including (but not limited to): 

```
Wrap every parameter
Wrap every argument
Indent all parameters
Indent all arguments
Align wrapped parameters
Align wrapped arguments
Indent wrapped parameters
Indent wrapped arguments
```

To make my life easier i took an approach of having loc strings like:

```
Wrap every {0}
Indent all {0}
Align wrapped {0}
Indent wrapped {0}
```

And then there are singular and plural forms of the thigns being wrapped.  i.e.:

```
Singular:
parameter
argument

Plural:
parameters
arguments
```

The singular nouns are used with `Wrap every {0}`, while the plural nouns are used with the other loc strings.

The question is: is this approach ok to mix/match in this way?  I could imagine, for example, in some language that there would need to be a special form for some of the combinations.  And it would not be ok to just jam the plural form of some noun into one of those loc forms.

Does this help clear things up?  It makes the code much easier.  But if it hurts loc in some languages then i'd prefer to do things "the right way" whatever that is.

Reply:
I see and understand the risk now. @tmeschter what do you think?

I don't speak enough languages to demonstrate a counter-example, but I would not be surprised to see one. I think the only thing that is "right" in the sense that it can't go wrong in this way is to just have separate resources for each of the "several strings we show user here".

How many strings are we talking about if we just split them out. Unless it's insane, I'd recommend just doing that and eliminating even having to worry about this.

@tmeschter What do you think?

Reply:
Saw above that you were going to proactively make the change, so +1 to that then :)

Reply:
Yeah, trying to compose localizable resources in this way is problematic and should be avoided if possible.

Reply:
> How many strings are we talking about if we split them out. Unless it's insane, I'd recommend doing that and eliminating even having to worry about this.

It will end up at about 4ish resources per construct we want to support.  Not something really problematic.

Thanks @nguerrera and @tmeschter !